tools/ci/get_version.py

```
@@ -0,0 +1,77 @@
+
+    version = ""
+    should_push_tag = False
+
+    if github_ref.startswith('refs/tags/'):
+        # 已由 tag 推送触发，直接使用该 tag 作为版本
+        version = github_ref[10:]
+    elif create_release:
+        # 手动触发且要求创建 release：生成新的 beta 版本
+        # 获取远程 tag 列表并按版本排序
+        cmd = ['git', 'ls-remote', '--refs', '--tags', '--sort=-version:refname', 'origin', 'v*']
+        result = subprocess.run(cmd, capture_output=True, text=True)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)